pkg/storage/chunk/storage/caching_index_client_test.go

```diff
@@ -105,11 +109,12 @@ func TestTempCachingStorageClient(t *testing.T) {
 		return true
 	})
 	require.NoError(t, err)
+	time.Sleep(1000 * time.Millisecond)
```

Comment:
Same suggestion to drop this and some of the similar instances below.

Reply:
> Same suggestion to drop this and some of the similar instances below.

done,thanks,  use assert.Eventually